module/proto/gravity/v1/msgs.proto

```diff
@@ -1,237 +1,176 @@
+// BatchTxSignature is a signature on behalf of a validator for a BatchTx.
+message BatchTxSignature {
+  string token_contract = 1;
+  uint64 nonce          = 2; 
+  string eth_signer     = 3;
+  bytes  signature      = 4 [
+    (gogoproto.casttype) = "github.com/ethereum/go-ethereum/common/hexutil.Bytes"
+  ];
 }
 
-message MsgConfirmLogicCallResponse {}
-
-// EthereumBridgeDepositClaim
-// When more than 66% of the active validator set has
-// claimed to have seen the deposit enter the ethereum blockchain coins are
-// issued to the Cosmos address in question
-// -------------
-message MsgDepositClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  string token_contract = 3;
-  string amount         = 4 [
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
-    (gogoproto.nullable)   = false
+// UpdateSignerSetTxSignature is a signature on behalf of a validator for a UpdateSignerSetTx
+message UpdateSignerSetTxSignature {
+  uint64 nonce                = 1;
+  string eth_signer           = 2;
+  bytes  signature            = 3 [
+    (gogoproto.casttype) = "github.com/ethereum/go-ethereum/common/hexutil.Bytes"
   ];
-  string ethereum_sender = 5;
-  string cosmos_receiver = 6;
-  string orchestrator    = 7;
 }
 
-message MsgDepositClaimResponse {}
+message MsgSubmitEthereumSignatureResponse {}
+
+// MsgSubmitEthereumEvent
+message MsgSubmitEthereumEvent {
+  option (gogoproto.goproto_getters) = false;
 
-// WithdrawClaim claims that a batch of withdrawal
-// operations on the bridge contract was executed.
-message MsgWithdrawClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  uint64 batch_nonce    = 3;
-  string token_contract = 4;
-  string orchestrator   = 5;
+  google.protobuf.Any event  = 1 [(cosmos_proto.accepts_interface) = "EthereumEvent"];
+  string              signer = 2;
 }
 
-message MsgWithdrawClaimResponse {}
-
-// ERC20DeployedClaim allows the Cosmos module
-// to learn about an ERC20 that someone deployed
-// to represent a Cosmos asset
-message MsgERC20DeployedClaim {
-  uint64 event_nonce    = 1;
-  uint64 block_height   = 2;
-  string cosmos_denom   = 3;
-  string token_contract = 4;
-  string name           = 5;
-  string symbol         = 6;
-  uint64 decimals       = 7;
-  string orchestrator   = 8;
+// SendToCosmosEvent is submitted when the SendToCosmosEvent is emitted by they gravity contract. 
+// ERC20 representation coins are minted to the cosmosreceiver address.
+message SendToCosmosEvent {
+  uint64 event_nonce     = 1;
+  string token_contract  = 2;
+  string amount          = 3 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", 
+    (gogoproto.nullable) = false
+  ];
+  string ethereum_sender = 4;
+  string cosmos_receiver = 5;
+  uint64 ethereum_height = 6;
+}
+
+// BatchTxExecutedEvent claims that a batch of BatchTxExecutedal operations on the bridge contract
+// was executed successfully on ETH
+message BatchTxExecutedEvent {
```

Comment:
It might be a good idea to change this naming scheme slightly, since some of these become a moutful

BatchTxExecutedEvent -> BatchExecutedEvent
ContractCallTxExecutedEvent -> ContractCallExecutedEvent

Because if you think about it, the Tx is the thing that causes something to be executed on Ethereum, but the event is not about the Tx, it is about what was executed on Ethereum

Reply:
Good call @jtremback so moved.